client/helpers/index.js

```diff
@@ -145,3 +145,45 @@ export const stripInvalidXmlChars = text => {
   }
   return text.replace(/([^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFC\u{10000}-\u{10FFFF}])/ug, '')
 };
+
+// protocol 1 and protocol 2
+// protocol 1, protocol 2 and protocol 3
+function getProtocolParts(protocols) {
+  return protocols.map((p, i) => {
+    const str = `${p.index + 1}`;
+    if (i === protocols.length - 1) {
+      return ` and ${str}`
+    }
+    if (i > 0) {
+      return `, ${str}`;
+    }
+    return str;
+  }).join('');
+}
+
+export const confirmRemove = (protocolRef, singularName, key) => (project, item) => {
+  const protocols = (project.protocols || [])
+    .map((protocol, index) => ({ ...protocol, index }))
+    .filter(protocol => {
+      return (protocol[protocolRef] || []).includes(key ? item[key] : item);
+    });
+
+  // item doesn't appear in any protocols
+  if (!protocols.length) {
+    return true;
+  }
+
+  let message = `Protocols will be affected\n\nRemoving this ${singularName} will also remove it from`;
+
+  // item appears in all protocols
+  if (protocols.length === (project.protocols || []).length) {
+    return window.confirm(`${message} all protocols.`)
+  }
+
+  const protocolText = protocols.length === 1
+    // item appears in a single protocol
+    ? `protocol ${protocols[0].index + 1}`
+    // item appears in many protocols, list them
+    : `protocols ${getProtocolParts(protocols)}`;
+  return window.confirm(`${message} ${protocolText}.`);
+};
```

Comment:
Could we split some of "helpers" out into individual files? Massive dumping ground index.js files are kind of a pain to work with.

Suggest creating `confirm-remove.js` and putting the green bits of this diff into it.